hazelcast/src/test/java/com/hazelcast/genericrecord/AbstractGenericRecordTest.java

```diff
@@ -0,0 +1,317 @@
+
+        HazelcastInstance[] instances = createCluster();
+
+        HazelcastInstance instance = createAccessorInstance(serializationConfig);
+        IMap<Object, Object> map = instance.getMap("test");
+        NamedPortable expected = new NamedPortable("foo", 900);
+
+        String key = generateKeyOwnedBy(instances[0]);
+        map.put(key, expected);
+        Object returnValue = map.executeOnKey(key, (EntryProcessor<Object, Object, Object>) entry -> {
+            Object value = entry.getValue();
+            GenericRecord genericRecord = (GenericRecord) value;
+
+            GenericRecord modifiedGenericRecord = genericRecord.newBuilder()
+                    .writeUTF("name", "bar")
+                    .writeInt("myint", 4).build();
+
+            entry.setValue(modifiedGenericRecord);
+
+            return genericRecord.readInt("myint");
+        });
+        assertEquals(expected.myint, returnValue);
+
+        NamedPortable actualPortable = (NamedPortable) map.get(key);
+        assertEquals("bar", actualPortable.name);
+        assertEquals(4, actualPortable.myint);
+    }
+
+    @Test
+    public void testCloneWithGenericBuilderOnEntryProcessor() throws IOException {
+
+        HazelcastInstance[] instances = createCluster();
+
+        HazelcastInstance instance = createAccessorInstance(serializationConfig);
+        IMap<Object, Object> map = instance.getMap("test");
+        NamedPortable expected = new NamedPortable("foo", 900);
+
+        String key = generateKeyOwnedBy(instances[0]);
+        map.put(key, expected);
+        Object returnValue = map.executeOnKey(key, (EntryProcessor<Object, Object, Object>) entry -> {
+            Object value = entry.getValue();
+            GenericRecord genericRecord = (GenericRecord) value;
+
+            GenericRecord modifiedGenericRecord = genericRecord.cloneWithBuilder()
+                    .writeInt("myint", 4).build();
+
+            entry.setValue(modifiedGenericRecord);
+
+            return genericRecord.readInt("myint");
+        });
+        assertEquals(expected.myint, returnValue);
+
+        NamedPortable actualPortable = (NamedPortable) map.get(key);
+        assertEquals("foo", actualPortable.name);
+        assertEquals(4, actualPortable.myint);
+    }
+
+    private static class GetInt implements Callable<Integer>, HazelcastInstanceAware, Serializable {
+
+        volatile HazelcastInstance instance;
+
+        @Override
+        public void setHazelcastInstance(HazelcastInstance hazelcastInstance) {
+            instance = hazelcastInstance;
+        }
+
+        @Override
+        public Integer call() throws Exception {
+            IMap<Object, Object> map = instance.getMap("test");
+            GenericRecord genericRecord = (GenericRecord) map.get(1);
+            return genericRecord.readInt("myint");
+        }
+    }
+
+    @Test
+    public void testRemoteLogicReturnsGenericRecord() throws Exception {
```

Comment:
test name misleading, the GetInt return Integer?

Reply:
The test is actually testing the `GenericRecord` is passed to `GetInt` when it calls map.get. Renaming it to `testGenericRecordIsReturnedInRemoteLogic`